# 📖 Exploración del Dataset
## Dataset: Student Performance Factors

Este proyecto analiza factores que influyen en el rendimiento estudiantil usando un dataset público de Kaggle.  
El objetivo de este Notebook es aplicar técnicas de **carga** y **exploración** del dataset.

## 🔧 1. Carga de librerías y dataset
En esta sección importamos las librerías necesarias y cargamos el dataset desde la carpeta `../data/`.

In [1]:
# Importar librerías
import pandas as pd

In [2]:
# Cargar el dataset
df = pd.read_csv("../data/student_performance.csv")

## 🔍 2. Exploración inicial
Aquí revisamos la forma del dataset, sus columnas y tipos de datos.

In [3]:
print("Primer vistazo al dataset:")
df.head()

Primer vistazo al dataset:


Hours_Studied  Attendance Parental_Involvement Access_to_Resources  \
0             23          84                  Low                High   
1             19          64                  Low              Medium   
2             24          98               Medium              Medium   
3             29          89                  Low              Medium   
4             19          92               Medium              Medium   

  Extracurricular_Activities  Sleep_Hours  Previous_Scores Motivation_Level  \
0                         No            7               73              Low   
1                         No            8               59              Low   
2                        Yes            7               91           Medium   
3                        Yes            8               98           Medium   
4                        Yes            6               65           Medium   

  Internet_Access  Tutoring_Sessions Family_Income Teacher_Quality  \
0             Yes                  0           Low          Medium   
1             Yes                  2        Medium          Medium   
2             Yes                  2        Medium          Medium   
3             Yes                  1        Medium          Medium   
4             Yes                  3        Medium            High   

  School_Type Peer_Influence  Physical_Activity Learning_Disabilities  \
0      Public       Positive                  3                    No   
1      Public       Negative                  4                    No   
2      Public        Neutral                  4                    No   
3      Public       Negative                  4                    No   
4      Public        Neutral                  4                    No   

  Parental_Education_Level Distance_from_Home  Gender  Exam_Score  
0              High School               Near    Male          67  
1                  College           Moderate  Female          61  
2             Postgraduate               Near    Male          74  
3              High School           Moderate    Male          71  
4                  College               Near  Female          70

In [4]:
print("Forma del dataset:", df.shape)
print("\nColumnas:")
print(df.columns)

Forma del dataset: (6607, 20)

Columnas:
Index(['Hours_Studied', 'Attendance', 'Parental_Involvement',
       'Access_to_Resources', 'Extracurricular_Activities', 'Sleep_Hours',
       'Previous_Scores', 'Motivation_Level', 'Internet_Access',
       'Tutoring_Sessions', 'Family_Income', 'Teacher_Quality', 'School_Type',
       'Peer_Influence', 'Physical_Activity', 'Learning_Disabilities',
       'Parental_Education_Level', 'Distance_from_Home', 'Gender',
       'Exam_Score'],
      dtype='object')


In [5]:
# Información general
df.info()

# Estadísticas descriptivas
df.describe(include='all') # Con include='all' hacemos que lo haga con cada columna

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6607 entries, 0 to 6606
Data columns (total 20 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Hours_Studied               6607 non-null   int64 
 1   Attendance                  6607 non-null   int64 
 2   Parental_Involvement        6607 non-null   object
 3   Access_to_Resources         6607 non-null   object
 4   Extracurricular_Activities  6607 non-null   object
 5   Sleep_Hours                 6607 non-null   int64 
 6   Previous_Scores             6607 non-null   int64 
 7   Motivation_Level            6607 non-null   object
 8   Internet_Access             6607 non-null   object
 9   Tutoring_Sessions           6607 non-null   int64 
 10  Family_Income               6607 non-null   object
 11  Teacher_Quality             6529 non-null   object
 12  School_Type                 6607 non-null   object
 13  Peer_Influence              6607 non-null   obje

Hours_Studied   Attendance Parental_Involvement Access_to_Resources  \
count     6607.000000  6607.000000                 6607                6607   
unique            NaN          NaN                    3                   3   
top               NaN          NaN               Medium              Medium   
freq              NaN          NaN                 3362                3319   
mean        19.975329    79.977448                  NaN                 NaN   
std          5.990594    11.547475                  NaN                 NaN   
min          1.000000    60.000000                  NaN                 NaN   
25%         16.000000    70.000000                  NaN                 NaN   
50%         20.000000    80.000000                  NaN                 NaN   
75%         24.000000    90.000000                  NaN                 NaN   
max         44.000000   100.000000                  NaN                 NaN   

       Extracurricular_Activities  Sleep_Hours  Previous_Scores  \
count                        6607   6607.00000      6607.000000   
unique                          2          NaN              NaN   
top                           Yes          NaN              NaN   
freq                         3938          NaN              NaN   
mean                          NaN      7.02906        75.070531   
std                           NaN      1.46812        14.399784   
min                           NaN      4.00000        50.000000   
25%                           NaN      6.00000        63.000000   
50%                           NaN      7.00000        75.000000   
75%                           NaN      8.00000        88.000000   
max                           NaN     10.00000       100.000000   

       Motivation_Level Internet_Access  Tutoring_Sessions Family_Income  \
count              6607            6607        6607.000000          6607   
unique                3               2                NaN             3   
top              Medium             Yes                NaN           Low   
freq               3351            6108                NaN          2672   
mean                NaN             NaN           1.493719           NaN   
std                 NaN             NaN           1.230570           NaN   
min                 NaN             NaN           0.000000           NaN   
25%                 NaN             NaN           1.000000           NaN   
50%                 NaN             NaN           1.000000           NaN   
75%                 NaN             NaN           2.000000           NaN   
max                 NaN             NaN           8.000000           NaN   

       Teacher_Quality School_Type Peer_Influence  Physical_Activity  \
count             6529        6607           6607        6607.000000   
unique               3           2              3                NaN   
top             Medium      Public       Positive                NaN   
freq              3925        4598           2638                NaN   
mean               NaN         NaN            NaN           2.967610   
std                NaN         NaN            NaN           1.031231   
min                NaN         NaN            NaN           0.000000   
25%                NaN         NaN            NaN           2.000000   
50%                NaN         NaN            NaN           3.000000   
75%                NaN         NaN            NaN           4.000000   
max                NaN         NaN            NaN           6.000000   

       Learning_Disabilities Parental_Education_Level Distance_from_Home  \
count                   6607                     6517               6540   
unique                     2                        3                  3   
top                       No              High School               Near   
freq                    5912                     3223               3884   
mean                     NaN                      NaN                NaN   
std                 

## 📋 Primeras observaciones 

- El dataset contiene **6607 filas** y **20 columnas**. 

- Las columnas incluyen tanto **variables numéricas** (ej. `Exam_Score`, `Hours_Studied`) como **categóricas** (ej. `Gender`, `Parental_Involvement`, `Motivation_Level`). 

### 2.1 Duplicados

In [6]:
# Comprobación de duplicados
duplicados = df.duplicated().sum()
print(f"Número de filas duplicadas: {duplicados}")

Número de filas duplicadas: 0


### 2.2 Valores nulos

In [7]:
# Comprobación de valores nulos
df.isna().sum()

Hours_Studied                  0
Attendance                     0
Parental_Involvement           0
Access_to_Resources            0
Extracurricular_Activities     0
Sleep_Hours                    0
Previous_Scores                0
Motivation_Level               0
Internet_Access                0
Tutoring_Sessions              0
Family_Income                  0
Teacher_Quality               78
School_Type                    0
Peer_Influence                 0
Physical_Activity              0
Learning_Disabilities          0
Parental_Education_Level      90
Distance_from_Home            67
Gender                         0
Exam_Score                     0
dtype: int64

### 2.3 Tipos de datos

En el dataset, se identificaron 13 columnas tipo objets con valores como `"Yes/No"`, `"Low/Medium/High"` o `"Public/Private"`.  

Columnas:

- `Parental_Involvement`
- `Access_to_Resources`
- `Extracurricular_Activities`
- `Motivation_Level`
- `Internet_Access`
- `Family_Income`
- `Teacher_Quality`
- `School_Type`
- `Peer_Influence`
- `Learning_Disabilities`
- `Parental_Education_Level`
- `Distance_from_Home`
- `Gender`

## 📋 Observaciones para la Limpieza

- Los nombres de las columnas ya estaban en `snake_case`, por lo que solo se convertirán a **minúsculas** para mantener coherencia.

- Comprobación de duplicados. El resultado fue **0 filas duplicadas**, por lo tanto **no será necesario eliminar registros repetidos**.

- Se detectaron valores *nulos* en **tres columnas** del dataset:

    - `Teacher_Quality` (78 nulos): **categórica**.
    - `Parental_Education_Level` (90 nulos): **categórica**.
    - `Distance_from_Home` (67 nulos): **categórica**.

➡️ No se realizará imputación de nulos. Las visualizaciones afectadas mostrarán únicamente los registros completos.

➡️ La validación de **rangos** y **categorías** se realizará en el notebook **2_transformation.ipynb**, como parte de la limpieza estructural.

Estas observaciones confirman que el dataset requiere una limpieza estructurada, que se abordará en el notebook `2_transformation.ipynb`.